addons/account/models/account_move_line.py

```diff
@@ -0,0 +1,2574 @@
+                'product'
+            ) if line.move_id.is_invoice() else 'product'
+
+    # Do not depend on `move_id.partner_id`, the inverse is taking care of that
+    def _compute_partner_id(self):
+        for line in self:
+            line.partner_id = line.move_id.partner_id.commercial_partner_id
+
+    @api.depends('move_id.date', 'move_id.currency_id')
+    def _compute_currency_id(self):
+        for line in self:
+            if line.display_type == 'cogs':
+                line.currency_id = line.company_currency_id
+            elif line.move_id.is_invoice(include_receipts=True):
+                line.currency_id = line.move_id.currency_id
+            else:
+                line.currency_id = line.currency_id or line.company_id.currency_id
+
+    @api.depends('product_id', 'journal_id')
+    def _compute_name(self):
+        for line in self:
+            if not line.product_id or line.display_type in ('line_section', 'line_note'):
+                continue
+            # YTI : Coming from conflicting onchange from account.move
+            # for line in self.filtered(lambda line: line.account_id.user_type_id.type in ('receivable', 'payable')):
+            #     line.name = self.payment_reference or ''
+            if line.partner_id.lang:
+                product = line.product_id.with_context(lang=line.partner_id.lang)
+            else:
+                product = line.product_id
+
+            values = []
+            if product.partner_ref:
+                values.append(product.partner_ref)
+            if line.journal_id.type == 'sale':
+                if product.description_sale:
+                    values.append(product.description_sale)
+            elif line.journal_id.type == 'purchase':
+                if product.description_purchase:
+                    values.append(product.description_purchase)
+            line.name = '\n'.join(values)
+
+    @api.depends('display_type', 'company_id')
+    def _compute_account_id(self):
+        term_lines = self.filtered(lambda line: line.display_type == 'payment_term')
+        if term_lines:
+            moves = term_lines.move_id
+            self.env.cr.execute(f"""
+                WITH previous AS (
+                    SELECT DISTINCT ON (line.move_id)
+                           'account.move' AS model,
+                           line.move_id AS id,
+                           NULL AS account_type,
+                           line.account_id AS account_id
+                      FROM account_move_line line
+                     WHERE line.move_id = ANY(%(move_ids)s)
+                       AND line.display_type = 'payment_term'
+                       AND line.id != ANY(%(current_ids)s)
+                ),
+                properties AS(
+                    SELECT DISTINCT ON (property.company_id, property.name)
+                           'res.partner' AS model,
+                           SUBSTR(property.res_id, {len('res.partner') + 2})::integer AS id,
```

Comment:
Using dynamic SQL for that doesn't seem necessary. This should be a parameter like the rest. Or it should use utilities like `split_part` or `regexp_*` to generically process reference values.